README.rst

```diff
@@ -71,7 +71,7 @@ Or instead freeze your current virtualenv via requirements.txt and execute it an
 
 .. code-block:: bash
 
-    $ pex $(pip freeze) -o my_virtualenv.pex
```

Comment:
The original command line is correct. What version of Pex do you have?

For example:
```bash
# Set up a test venv to freeze into a PEX:
:; python -mvenv /tmp/test.venv
:; source /tmp/test.venv/bin/activate
:; pip -q install requests black pytest
:; pip freeze
black==24.4.2
certifi==2024.7.4
charset-normalizer==3.3.2
click==8.1.7
exceptiongroup==1.2.2
idna==3.7
iniconfig==2.0.0
mypy-extensions==1.0.0
packaging==24.1
pathspec==0.12.1
platformdirs==4.2.2
pluggy==1.5.0
pytest==8.2.2
requests==2.32.3
tomli==2.0.1
typing_extensions==4.12.2
urllib3==2.2.2

# Now do the freeze -> PEX:
:; pex -V
2.10.0
:; pex $(pip freeze) -o my_virtualenv.pex

# And it's all in there:
:; pex-tools my_virtualenv.pex info -i2
{
  "bootstrap_hash": "a83d6ee254ee17d37ef66f706a4ac703d2020cc6",
  "build_properties": {
    "pex_version": "2.10.0"
  },
  "code_hash": "c86bb06a71909b393d572d067c8c4248e479492d",
  "deps_are_wheel_files": false,
  "distributions": {
    "black-24.4.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl": "4701ca3bb6955e72f36fbd868da1f2f9135e38f25d66fcf651b6310723fcc28b",
    "certifi-2024.7.4-py3-none-any.whl": "f51955747a040ad60cc9b281d2b81748fda55a5bb5c4b6b10be65f0354609302",
    "charset_normalizer-3.3.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl": "8cd9bbbbbe1993cfd2a775037fb35ac3c3d4d1d3672da5ddcac24eade85fd078",
    "click-8.1.7-py3-none-any.whl": "8ab007356e466bf716f661c3ad7954130bcc92d6fa326ab3cf424a3dd5d83254",
    "exceptiongroup-1.2.2-py3-none-any.whl": "92cf3ff3567ada0936ec860ae517dab43320535889fcc0612bcc4cc40db274b9",
    "idna-3.7-py3-none-any.whl": "243156850dfb4ab00c4b9604d58d83e722e41eb0f445358feb23af5701231307",
    "iniconfig-2.0.0-py3-none-any.whl": "988e40b3b55723f6802760cae3cd0319699d3e69ae7f34c7793d38fb44d579b7",
    "mypy_extensions-1.0.0-py3-none-any.whl": "41a5b82d9710c42c0037b534f67c353e679e49f92df03db3e5e73ef236c1f44e",
    "packaging-24.1-py3-none-any.whl": "25baba288a0f0e2ebe1b0117cad380c5d4c0137aff4a2c7755f92cffd0eb38f8",
    "pathspec-0.12.1-py3-none-any.whl": "37babcedd95f7e3d43e2ca2561501ae74c525e280620fe3944329a4f357713f6",
    "platformdirs-4.2.2-py3-none-any.whl": "ab85a6d0ac2ac7ec1a1d66450604d93c62d0c8d30cfeee8487d39962a3a5caf4",
    "pluggy-1.5.0-py3-none-any.whl": "6826f70172987927d47205f5bc5b1b18c380201149e5b63bc90e4d1645e7a8d9",
    "pytest-8.2.2-py3-none-any.whl": "ce97a2c0ec29fd42f042f7e76e7f857fb83919d9054169b16988d628f47b43f4",
    "requests-2.32.3-py3-none-any.whl": "2f57f4ae5920a2f5f55a8f4971fe010b00cb2a30f8ab9c572826150fa309c450",
    "tomli-2.0.1-py3-none-any.whl": "8af30168c90e438c0488ddb0dcc8bfbdb7f54299570473943266c8f800de58dc",
    "typing_extensions-4.12.2-py3-none-any.whl": "ab87cc50d4264524df94030d479d1501922ac3d258a1d09cff936840137acd76",
    "urllib3-2.2.2-py3-none-any.whl": "568f866565573953519a3419f2ab438633b3fb6e932725d0a1aecb46a35d9b12"
  },
  "emit_warnings": true,
  "excluded": [],
  "ignore_errors": false,
  "includes_tools": false,
  "inherit_path": "false",
  "inject_args": [],
  "inject_env": {},
  "inject_python_args": [],
  "interpreter_constraints": [],
  "max_install_jobs": 1,
  "overridden": [],
  "pex_hash": "1bfd65ff6c5cff0e9e37bc48b0a3f58747f71ca6",
  "pex_path": "",
  "pex_paths": [],
  "requirements": [
    "black==24.4.2",
    "certifi==2024.7.4",
    "charset-normalizer==3.3.2",
    "click==8.1.7",
    "exceptiongroup==1.2.2",
    "idna==3.7",
    "iniconfig==2.0.0",
    "mypy-extensions==1.0.0",
    "packaging==24.1",
    "pathspec==0.12.1",
    "platformdirs==4.2.2",
    "pluggy==1.5.0",
    "pytest==8.2.2",
    "requests==2.32.3",
    "tomli==2.0.1",
    "typing_extensions==4.12.2",
    "urllib3==2.2.2"
  ],
  "strip_pex_env": true,
  "venv": false,
  "venv_bin_path": "false",
  "venv_copies": false,
  "venv_hermetic_scripts": true,
  "venv_site_packages_copies": false,
  "pex_root": "/home/jsirois/.pex"
}
```

However, using your command line, you do not get what you want [^1]:
```bash
:; pex -o my_virtualenv.pex -- $(pip freeze)
:; pex-tools my_virtualenv.pex info -i2
{
  "bootstrap_hash": "a83d6ee254ee17d37ef66f706a4ac703d2020cc6",
  "build_properties": {
    "pex_version": "2.10.0"
  },
  "code_hash": "c86bb06a71909b393d572d067c8c4248e479492d",
  "deps_are_wheel_files": false,
  "distributions": {},
  "emit_warnings": true,
  "excluded": [],
  "ignore_errors": false,
  "includes_tools": false,
  "inherit_path": "false",
  "inject_args": [],
  "inject_env": {},
  "inject_python_args": [],
  "interpreter_constraints": [],
  "max_install_jobs": 1,
  "overridden": [],
  "pex_hash": "281d2e032b8bcbb950660a555247946b3ac5080d",
  "pex_path": "",
  "pex_paths": [],
  "requirements": [],
  "strip_pex_env": true,
  "venv": false,
  "venv_bin_path": "false",
  "venv_copies": false,
  "venv_hermetic_scripts": true,
  "venv_site_packages_copies": false,
  "pex_root": "/home/jsirois/.pex"
}
```

[^1]: Arguably, `pex ... -o a.pex -- <any pass through args at all>` should fail, since Pex has nothing to pass through those arguments to; i.e.: `--` says, as in most programs, stop processing args here and pass the remaining off to some other (sub) process.

Reply:
Yeah, sop even going back to Pex 1.0.0 which was released I find the existing README advice works:
```bash
# Now with a smaller venv that has distributions very old Pex can handle:
:; pip freeze
ansicolors==1.1.8
cowsay==5.0

# A fancy way to run Pex 1.0.0 using current Pex (Pex has pipx-like functionality -
# when you do not specify `-o`, it just runs the (ephemeral) PEX immediately, passing
# all args after `--` to the ephemeral PEX - in this case a Pex PEX):
:; pex --python python2.7 pex==1.0.0 -c pex -- $(pip freeze) -o my_virtualenv.pex
:; unzip -qc my_virtualenv.pex PEX-INFO | jq .
{
  "requirements": [
    "cowsay==5.0",
    "ansicolors==1.1.8"
  ],
  "inherit_path": false,
  "distributions": {
    "ansicolors-1.1.8-py2.py3-none-any.whl": "b78871aaa4ca104db1a7a2c2dc2793554d1d5e01",
    "cowsay-5.0-py2.py3-none-any.whl": "3ebdcd27726ac31ecc29fcaa6735fb874249ccdb"
  },
  "always_write_cache": false,
  "code_hash": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
  "zip_safe": true,
  "ignore_errors": false,
  "build_properties": {
    "platform": "linux-x86_64",
    "version": [
      2,
      7,
      18
    ],
    "class": "CPython"
  }
}
```